pkg/kubelet/server/server.go

```diff
@@ -878,6 +890,39 @@ func (s *Server) getPortForward(request *restful.Request, response *restful.Resp
 	proxyStream(response.ResponseWriter, request.Request, url)
 }
 
+// checkpoint handles the checkpoint API request. It checks if the requested
+// podNamespace, pod and container actually exist and only then calls out
+// to the runtime to actually checkpoint the container.
+func (s *Server) checkpoint(request *restful.Request, response *restful.Response) {
+	if request == nil || response == nil {
+		return
+	}
+	pod, ok := s.host.GetPodByName(request.PathParameter("podNamespace"), request.PathParameter("podName"))
+	if !ok {
+		response.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	containerName := request.PathParameter("containerName")
+
+	found := false
+	for _, container := range pod.Spec.Containers {
+		if container.Name == containerName {
```

Comment:
Relying on just featureGate seems like a hammer flag to enable checkpoints for all containers running inside the cluster. Should there be an additional container level spec flag to enable checkpoints for containers of interest and default being false? Without a container level spec flag and with just featureGate enabled, all containers would have a specific checkpoint on the node and depending on number of containers running on the node it might create a disk pressure if enough disk space is not allocated. Any thoughts?